core/templates/hash_map.h

```diff
@@ -395,6 +395,32 @@ class HashMap : private Allocator {
 		return true;
 	}
 
+	// Renew the entry with the given key, making it the newest entry in insertion order.
+	// Equals to erase(key) followed by insert(key, value).
+	TValue *renew_key(const TKey &p_key) {
+		ERR_FAIL_COND_V(elements == nullptr || num_elements == 0, nullptr);
+		uint32_t pos = 0;
+		ERR_FAIL_COND_V(!_lookup_pos(p_key, pos), nullptr);
```

Comment:
Your lru currently prints errors where it didn't previously.
I'd also rename the function to show that it's doing the same as `getptr`, except for the "move to back" thing.

```suggestion
	TValue *getptr_and_move_to_back(const TKey &p_key) {
		uint32_t pos = 0;
		if (elements == nullptr || num_elements == 0 || !_lookup_pos(p_key, pos) {
			return nullptr;
		}
```

Reply:
I think `getptr_and_renew/refresh` is better because this is designed to avoid memory operation in `erase` followed by `insert`. `move_to_back` is not clear in where we move, might be `move_to_back_in_insertion_order` but that'd be too long.